.gitignore

```diff
@@ -0,0 +1,45 @@
+# Byte-compiled / optimized / DLL files
+__pycache__/*.*
+*/__pycache__/
+*.py[cod]
+*$py.class
+
+# PyCharm / Git folders
+.idea/
+
+# Script files
+src/main/scripts/
+
+# VS Code folders
+.vscode/
+
+# Unit test / coverage reports
+.coverage
+.coverage.*
+nosetests.xml
+coverage.xml
+.pytest_cache/
+
+# Sphinx documentation
+docs/_build/
+docs/_static/
+docs/.doctrees/
+docs/janusgraph_python*.rst
+docs/modules.rst
+docs/janusgraph_python.rst
+
+# PyBuilder
+target/
+
+# pyenv
+.python-version
+
+# Environments
+.env
+
+# mkdocs documentation
+site/
+*/site/
+
```

Comment:
I think the generated predicates should be added as well. That way when you build the library you won't have to do a git reset to pull new changes.

```
	src/main/python/janusgraph_python/core/attribute/GeoPredicate/*
	src/main/python/janusgraph_python/core/attribute/TextPredicate/*
```

Reply:
Can you clarify on this @chupman ? I mean why would you need to do git reset in that directory?  Also, I don't see that directory being present anymore. (i.e. GeoPredicate & TextPredicate dir). I've moved all TextPredicates under src/main/python/janusgraph-python/core/attribute/Text.py and GeoPredicates under src/main/python/janusgraph-python/core/attribute/Geo.py based on comments before. 

Then I don't see a point of adding directories into gitignore which don't exist. Maybe I didn't understand your comment clearly.